modules/basic/stream/byte_stream.vineyard-mod

```diff
@@ -153,8 +153,11 @@ class ByteStream : public Registered<ByteStream> {
    * @return The unique pointer to the reader
    */
   std::unique_ptr<ByteStreamReader> OpenReader(Client& client) {
-    return std::unique_ptr<ByteStreamReader>(
-        new ByteStreamReader(client, id_, meta_));
+    if (client.MarkStream(id_, 2).ok()) {
+      return std::unique_ptr<ByteStreamReader>(
+          new ByteStreamReader(client, id_, meta_));
+    }
+    return nullptr;
   }
```

Comment:
I suggest to have API like

```cpp
Status OpenReader(Client &client, std::unique_ptr<ByteStreamReader> &r) {
   RETURN_ON_ERROR(client.MarkStream(id_, 2))....
   r = ...
   return Status::OK();
}
```

which will be more consistent will the style of our other APIs, and more importantly, let user know why the operation fail. Return `nullptr` sliently usually the root of many unintended bugs.